src/monobase/build.py

```diff
@@ -147,13 +157,16 @@ def build_generation(args: argparse.Namespace, mg: MonoGen) -> None:
     suffix = '' if args.environment == 'prod' else f'-{args.environment}'
     rdir = os.path.join(HERE, f'requirements{suffix}', f'g{mg.id:05d}')
 
-    cuda_versions = desc_version(mg.cuda.keys())
+    cuda_versions = desc_version(cudas.keys())
     if args.mini:
         # Mini mono with Torch but without CUDA or CuDNN, use CPU Torch
-        if len(mg.torch) == 1 and len(mg.cuda) == 0:
+        if len(mg.torch) == 1 and len(cudas) == 0:
             cuda_versions = ['cpu']
+    elif args.no_cuda:
+        # CPU node, install CPU Torch only
+        cuda_versions = ['cpu']
     else:
-        # Production, always add CPU torch
+        # GPU node, install both GPU & CPU Torch
         cuda_versions = ['cpu'] + cuda_versions
```

Comment:
```suggestion
    else:
        # Production, always add CPU torch
        cuda_versions = ['cpu'] + cuda_versions
```

`cuda_versions` will be `[]`, won't it, because `cudas == {}` if `args.no_cuda`?

Reply:
No cuda == no GPU torch == CPU torch only

Reply:
Yes, sure, but we didn't need to repeat the logic did we?